File: src/api/applog.ts
import { invoke } from '@tauri-apps/api/core';

import type { LogLevel } from '../types';

export function readAppLog(): Promise<string> {
  return invoke('read_app_log');
}

export function clearAppLog(): Promise<void> {
  return invoke('clear_app_log');
}

export function appendAppLog(level: LogLevel, message: string): Promise<void> {
  return invoke('append_app_log', { level, message });
}

File: src-tauri/src/commands/applog.rs
use tauri::State;

use crate::invoke_error::InvokeErrorDto;
use crate::state::AppState;

#[tauri::command]
pub fn read_app_log(state: State<'_, AppState>) -> Result<String, InvokeErrorDto> {
    Ok(state.app_log.read())
}

#[tauri::command]
pub fn clear_app_log(state: State<'_, AppState>) -> Result<(), InvokeErrorDto> {
    state.app_log.clear();
    Ok(())
}

#[tauri::command]
pub fn append_app_log(state: State<'_, AppState>, level: String, message: String) -> Result<(), InvokeErrorDto> {
    let level = match level.as_str() {
        "ok" | "warn" | "error" | "info" => level.as_str(),
        _ => "info",
    };
    state.app_log.append(level, &message);
    Ok(())
}